google/cloud/bigquery/v2/minimal/internal/job_rest_stub.cc

```diff
@@ -62,6 +62,41 @@ StatusOr<GetJobResponse> DefaultBigQueryJobRestStub::GetJob(
   return GetJobResponse::BuildFromHttpResponse(*http_response);
 }
 
+StatusOr<ListJobsResponse> DefaultBigQueryJobRestStub::ListJobs(
+    rest_internal::RestContext& rest_context, ListJobsRequest const& request) {
+  internal::OptionsSpan span(options_);
+
+  if (request.project_id().empty()) {
+    return internal::InvalidArgumentError(
+        "Invalid ListJobsRequest: Project Id is empty", GCP_ERROR_INFO());
+  }
```

Comment:
This seems to relate to something I mentioned previously:

>> I have addeed input validation checks for the presence of all required parameters where we know the api will fail without them.
> 
> For what it's worth, this is the opposite of what I would suggest. That is, I would remove all input validation checks where we know the API will fail without them. It duplicates tests, possibly inventing new error values, and is fragile in the case of changing requirements in the backend.

So, unless we have some client-side code that depends on `request.project_id` being non-empty, we should avoid such checks.  Instead, just let the chips fall where they may.

Reply:
https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2018-06-19-do-not-duplicate-server-side-validation.md